Documentation/Bookmarks/README.md

```diff
@@ -0,0 +1,71 @@
+a bookmark and perform some action.
+
+## Workflow
+
+The app can define the `BookmarkView` in its view hierarchy, as `BookmarkView` extends FrameLayout. The app 
+has to set the `bookmarks` property on the `BookmarkView` with the map's bookmarks. To handle the event when the 
+user taps on an item in the list, the app has to implement `BookmarkView.onItemClickListener` interface.
+
+Here is an example XML code that shows the BookMarkView below a MapView and the BookmarkView's bookmarks attribute 
+being bound to `map.bookmarks` via mapViewModel's bookmarks property:
+
+```
+  <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent">
+
+        <com.esri.arcgisruntime.mapping.view.MapView
+                android:id="@+id/mapView"
+                android:layout_width="0dp"
+                android:layout_height="0dp"
+                app:layout_constraintBottom_toTopOf="@id/guideline"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent"
+                app:map="@{mapViewModel.map}">
+        </com.esri.arcgisruntime.mapping.view.MapView>
+
+        <com.esri.arcgisruntime.toolkit.bookmark.BookmarkView
+                android:id="@+id/bookmarkView"
+                android:layout_width="0dp"
+                android:layout_height="0dp"
+                app:bookmarks="@{mapViewModel.bookmarks}"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/guideline" />
+
+        <androidx.constraintlayout.widget.Guideline
+                android:id="@+id/guideline"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:orientation="horizontal"
+                app:layout_constraintGuide_percent="0.5" />
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+```
+
+Here is example Kotlin code to set the activity that implements the BookmarkView's onItemClickListener interface as the 
+onItemClickListener and implementing the `onItemClick()`
+
+```
+class BookmarkActivity : AppCompatActivity(), BookmarkView.OnItemClickListener<Bookmark> {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        ...
+        ...
+        
+        bookmarkView.onItemClickListener = this
+    }
+
+    override fun onItemClick(item: Bookmark) {
+        mapView.setViewpointAsync(item.viewpoint)
+    }
+
+```
+
+To see it in action, try out the bookmark test in the [toolkit-test-app](https://github.com/Esri/arcgis-runtime-toolkit-android/tree/master/toolkit-test-app/src/main/java/com/esri/arcgisruntime/toolkit/test)
```

Comment:
Instead of copying source code here in the doc, would it make more sense to link to various parts of the toolkit test app instead? My concern is that this code in the doc can go out of date if the actual implementation changes.

Something like:

For an example of how to use a `BookmarkView` in a layout, see the [bookmark activity layout](https://github.com/Esri/arcgis-runtime-toolkit-android/blob/develop/toolkit-test-app/src/main/res/layout/activity_bookmark.xml)

For an example of how to handle click events from a `BookmarkView` see...

Reply:
I guess we can do that, but I thought the reason for having code in the doc so the user can quickly refer to it while reading the doc. We do point them to see the entire test-app.

 >My concern is that this code in the doc can go out of date if the actual implementation changes.   

I agree that updating this doc will be an overhead if in case the implementation changes. But not having the code in the doc and just point to the source code will not lead to a good user experience of having the code snippet right there. Right now we are consistent with iOS in how the documentation is presented https://github.com/Esri/arcgis-runtime-toolkit-ios/tree/master/Documentation/Bookmarks.  
If you think moving forward we should not have code snippets in the doc, I am happy to remove this here.

Reply:
Chipping in from the perspective of the original author of the doc for Scalebar https://github.com/Esri/arcgis-runtime-toolkit-android/wiki/Scalebar and Compass https://github.com/Esri/arcgis-runtime-toolkit-android/wiki/Compass...

If the snippets are as small as the ones there (I'm thinking especially of the 2-5 line snippets of Kotlin and Java) then I think it's best for them to be in the doc. It'd take a significant change to the API to make these out of date and I expect us to review the doc if we change the API.

Reply:
I agree with @alan-edi that code snippets are ok as long as they are kept to a minimum. Could we make the snippets of the layout xml smaller? I don't think we need the `MapView` and `Guideline` definitions in that xml. However, I think you would need the data binding declaration in the xml if you want to show data binding in the snippet:
```
<data>

        <variable
                name="mapViewModel"
                type="com.esri.arcgisruntime.toolkit.test.bookmark.map.MapViewModel" />
        </data>
```

Reply:
done.